src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -20,9 +20,9 @@ internal static class GoToDefinitionHelpers
             ISymbol symbol,
             Project project,
             IEnumerable<Lazy<INavigableItemsPresenter>> presenters,
-            ITypeSymbol containingTypeSymbol,
-            bool throwOnHiddenDefinition,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
+            bool thirdPartyNavigationAllowed = true,
+            bool throwOnHiddenDefinition = false)
```

Comment:
No, it doesn't enforce it if there are optional parameters and it itself is not optional.  See https://github.com/dotnet/roslyn-analyzers/blob/master/src/Roslyn/Core/ApiDesign/CancellationTokenMustBeLastAnalyzer.cs if you want.